File: frontend/types/analytics.ts
export interface AnalyticsSummary {
  calls_today: number;
  calls_change: number;
  cache_hit_percent: number;
  avg_response_ms: number;
  response_change_ms: number;
}

export interface UsageMetrics {
  total_calls: number;
  successful_calls: number;
  failed_calls: number;
  cache_hits: number;
  cache_hit_rate: number;
  avg_response_time_ms: number;
  estimated_cost_saved: number;
}

export interface ServiceBreakdown {
  service_name: string;
  call_count: number;
  cache_hits: number;
  avg_response_time_ms: number;
  error_count: number;
}

export interface AnalyticsResponse {
  period: string;
  metrics: UsageMetrics;
  by_service: ServiceBreakdown[];
  generated_at: string;
}

export interface UsageDataPoint {
  time: string;
  requests: number;
}

export interface ActivityLog {
  id: string;
  status: number;
  statusLabel: string;
  method: string;
  endpoint: string;
  latency: number;
  timestamp: string;
}

File: backend/app/schemas/analytics.py
"""Analytics and usage reporting schemas."""

from pydantic import BaseModel, Field


class UsageMetrics(BaseModel):
    """Aggregated usage metrics for a time period."""

    total_calls: int
    successful_calls: int
    failed_calls: int
    cache_hits: int
    cache_hit_rate: float = Field(ge=0.0, le=1.0)
    avg_response_time_ms: float
    estimated_cost_saved: float


class ServiceBreakdown(BaseModel):
    """Per-service usage breakdown."""

    service_name: str
    call_count: int
    cache_hits: int
    avg_response_time_ms: float
    error_count: int


class AnalyticsResponse(BaseModel):
    """Full analytics response for a dashboard period."""

    period: str
    metrics: UsageMetrics
    by_service: list[ServiceBreakdown]
    generated_at: str
